### PR TITLE
Segment metrics

### LINKED ISSUE #4: Add segment.io

### PR DESCRIPTION
**Pull request by jordanh**
Closes #4.

Currently implemented to track:

- User identity
- Login events
- Logout events
- Page views

### CODE REVIEW

**Comment by jordanh** [comment]
@mattkrick I'm pleased as punch with the way this integrated.

Curious for your thoughts and 👀  on:

1. Using `redux-segment` and action meta as the pattern to report to segment.io
2. The way I integrated this snippet in `Html.js`
3. The way I update page location in segment using React lifecycle (read that in a Medium article)
4. The way I clear the segment analytics identity information in the logout container

File: src/server/Html.js [hunk 1]
```diff
@@ -34,6 +35,18 @@ export default function Html({
         <meta property="description" content="Team transparency, made easy." />
         <style dangerouslySetInnerHTML={{__html: lookCSSToken}} id={lookConfig.styleElementId} />
         <title>{title}</title>
+        {/* segment.io analytics */}
+        {process.env.SEGMENT_WRITE_KEY &&
```
Comment (mattkrick) [inline src/server/Html.js]: Let's not do this in runtime. Let's create a static file & just load that.

File: src/universal/components/Action/Action.js [hunk 2]
```diff
@@ -15,6 +15,13 @@ export default class Action extends Component {
     children: PropTypes.any
   };
 
+  componentDidUpdate() {
+    if (window) {
```
Comment (mattkrick) [inline src/universal/components/Action/Action.js]: `if (typeof window !== 'undefined')`, right?

File: src/server/Html.js (hunk 1 above)
Comment (jordanh) [inline src/server/Html.js]: How about a compromise and I make this a react component?

File: src/universal/components/Action/Action.js (hunk 2 above)
Comment (jordanh) [inline src/universal/components/Action/Action.js]: Doh!

File: src/server/Html.js (hunk 1 above)
Comment (mattkrick) [inline src/server/Html.js]: Not sure i follow, We'll still be building and minifying the same thing for each visitor, right?

Comment (mattkrick) [inline src/server/Html.js]: Similar to what we do with the webpack manifest and themes, I'm thinking we build a static text file and just import from there

Comment (jordanh) [inline src/server/Html.js]: I got what you meant, I just didn't want to have to go and create _another_ build task. The thing is, you're 100% right. I'll do the right thing 💪

File: src/universal/containers/Logout/LogoutContainer.js [hunk 3]
```diff
@@ -26,6 +27,11 @@ export default class LogoutContainer extends Component {
     router.replace('/');
     dispatch(resetAppState());
     dispatch(showSuccess(logoutSuccess));
+    dispatch(segmentEvent('User Logout'));
+    if (window) {
```
Comment (mattkrick) [inline src/universal/containers/Logout/LogoutContainer.js]: this one too

File: src/server/Html.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 /* eslint react/no-danger:0 */
 import React, {PropTypes} from 'react';
-import { LookRoot } from 'react-look';
+import {LookRoot} from 'react-look';
 import {Provider} from 'react-redux';
 import {RouterContext} from 'react-router';
 import {renderToString} from 'react-dom/server';
 
+const segmentSnippet = require('../../build/segmentSnippet.json');
```
Comment (jordanh) [inline src/server/Html.js]: We cool now homey? 😎  @mattkrick

Comment (jordanh) [inline src/server/Html.js]: After sleeping on this implementation – I'm not sure I'm in love with it.

This doesn't really have any advantages over calculating the snippet once in memory (when declared in `Html.js`'s global scope.

I am going to take one more turn on this...

File: src/server/Html.js (hunk 1 above)
Comment (jordanh) [inline src/server/Html.js]: @mattkrick you'll note that I did it as a file include in https://github.com/ParabolInc/action/pull/250/commits/eae4e9627ff26734a60c5c5158c34055498cd449 and slept on it.

The things I didn't quite like were:

-  Importing the pre-computed snipped source into the module global scope was computationally equal to calling the snippet generation function. Either it happens before the server starts or it happens once after the server starts. Same difference AFAIAC.
 - Yet another mystery json file
 - Unless I introduced a branch (a la process.env... &&) there wasn't a great way to omit _some_ tag generation altogeter

I think the approach I've got now is decent.

File: webpack/webpack.config.dev.js [hunk 5]
```diff
@@ -86,7 +86,6 @@ export default {
   module: {
     loaders: [
       {test: /\.json$/, loader: 'json-loader'},
-      {test: /\.txt$/, loader: 'raw-loader'},
```
Comment (jordanh) [inline webpack/webpack.config.dev.js]: We weren't using this, and it wasn't working (I'm not sure why). So, I removed it.

**Comment by jordanh** [comment]
Ok, I think this is GTG now @mattkrick, do you concur, sir?

**Pull request merged**